recipes/asio-grpc/all/conanfile.py

```diff
@@ -56,17 +63,18 @@ def configure(self):
             raise ConanInvalidConfiguration(f"{self.name} 'recycling_allocator' cannot be used in combination with the 'unifex' backend.")
 
     def requirements(self):
-        self.requires("grpc/1.54.3", transitive_libs=True)
-        if self._local_allocator_option == "boost_container" or self.options.backend == "boost":
+        self.requires("grpc/1.54.3", transitive_headers=True, transitive_libs=True)
+        if (self._local_allocator_option == "boost_container" and Version(self.version) < "3.0.0") or self.options.backend == "boost":
             self.requires("boost/1.83.0", transitive_headers=True)
         if self.options.backend == "asio":
             self.requires("asio/1.29.0", transitive_headers=True)
         if self.options.backend == "unifex":
-            self.requires("libunifex/0.4.0")
+            self.requires("libunifex/0.4.0", transitive_headers=True, transitive_libs=True)
 
     def package_id(self):
         self.info.clear()
-        self.info.options.local_allocator = self._local_allocator_option
+        if self._local_allocator_option:
+            self.info.options.local_allocator = self._local_allocator_option
```

Comment:
Why does the generated package depend on the allocator option? I'd like a tad more insight into this :)

Reply:
Instead of using preprocessor macros, older version of asio-grpc were actually generating source files at build time. E.g. when local allocator is boost.container it would generate a header that uses it and install only that. If local allocator is std::memory_resource the same thing. See here https://github.com/Tradias/asio-grpc/blob/v2.9.3/src/CMakeLists.txt#L33-L45 .